bump org.gaul:modernizer-maven-plugin to 3.5.0

from https://github.com/checkstyle/checkstyle/pull/20938

```
[INFO] --- modernizer:3.5.0:modernizer (modernizer) @ checkstyle ---
[ERROR] /Users/runner/work/1/s/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:366: Prefer java.lang.String.split(String); note that split takes a regular expression rather than a set of delimiter characters
[ERROR] /Users/runner/work/1/s/src/main/java/com/puppycrawl/tools/checkstyle/filters/CsvFilterElement.java:50: Prefer java.lang.String.split(String); note that split takes a regular expression rather than a set of delimiter characters
[ERROR] /Users/runner/work/1/s/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:495: Prefer java.lang.String.split(String); note that split takes a regular expression rather than a set of delimiter characters
[ERROR] /Users/runner/work/1/s/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java:773: Prefer java.lang.String.split(String); note that split takes a regular expression rather than a set of delimiter characters
[ERROR] /Users/runner/work/1/s/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java:459: Prefer java.lang.String.split(String); note that split takes a regular expression rather than a set of delimiter characters

```

--- a/pom.xml
+++ b/pom.xml
@@ -904,7 +904,7 @@
       <plugin>
         <groupId>org.gaul</groupId>
         <artifactId>modernizer-maven-plugin</artifactId>
-        <version>3.4.0</version>
+        <version>3.5.0</version>
         <configuration>
           <javaVersion>${java.version}</javaVersion>
           <includeTestClasses>false</includeTestClasses>

--- a/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java
@@ -25,7 +25,6 @@ import java.net.URI;
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
-import java.util.StringTokenizer;
 import java.util.regex.Pattern;
 
 import javax.annotation.Nullable;
@@ -362,12 +361,13 @@ public abstract class AbstractAutomaticBean
         @Override
         @SuppressWarnings("unchecked")
         public Object convert(Class type, Object value) {
-            final StringTokenizer tokenizer = new StringTokenizer(
-                    value.toString(), COMMA_SEPARATOR);
+            final String[] tokens = value.toString().split(COMMA_SEPARATOR, -1);
             final List<Pattern> result = new ArrayList<>();
 
-            while (tokenizer.hasMoreTokens()) {
-                final String token = tokenizer.nextToken();
+            for (String token : tokens) {
+                if (token.isEmpty()) {
+                    continue;
+                }
                 result.add(CommonUtil.createPattern(token.trim()));
             }
 
@@ -455,12 +455,13 @@ public abstract class AbstractAutomaticBean
         @Override
         @SuppressWarnings("unchecked")
         public Object convert(Class type, Object value) {
-            final StringTokenizer tokenizer = new StringTokenizer(
-                value.toString().trim(), COMMA_SEPARATOR);
+            final String[] tokens = value.toString().trim().split(COMMA_SEPARATOR, -1);
             final List<String> result = new ArrayList<>();
 
-            while (tokenizer.hasMoreTokens()) {
-                final String token = tokenizer.nextToken();
+            for (String token : tokens) {
+                if (token.isEmpty()) {
+                    continue;
+                }
                 result.add(token.trim());
             }
 
@@ -490,13 +491,13 @@ public abstract class AbstractAutomaticBean
         @Override
         @SuppressWarnings("unchecked")
         public Object convert(Class type, Object value) {
-            // Converts to a String and trims it for the tokenizer.
-            final StringTokenizer tokenizer = new StringTokenizer(
-                value.toString().trim(), COMMA_SEPARATOR);
+            final String[] tokens = value.toString().trim().split(COMMA_SEPARATOR, -1);
             final List<AccessModifierOption> result = new ArrayList<>();
 
-            while (tokenizer.hasMoreTokens()) {
-                final String token = tokenizer.nextToken();
+            for (String token : tokens) {
+                if (token.isEmpty()) {
+                    continue;
+                }
                 result.add(AccessModifierOption.getInstance(token));
             }
 

